lib/scihist_digicoll/task_helpers/subject_creator_adjuster.rb

```diff
@@ -0,0 +1,56 @@
+module ScihistDigicoll
+  module TaskHelpers
+
+    # This is meant to be called by the rake task scihist:data_fixes:adjust_subjects_and_creators
+    # which instantiates this lass *once*, then calls process_work(work) on each work in the collection.
+    # process_work just updates the subjects and authors of a work according to @metadata_map.
+    # The goal is to separate the tasks of determining whether a work needs to be changed and how to change it
+    # (here) from the task of iterating over all works.
+    class SubjectCreatorAdjuster
+      attr_accessor :metadata_map, :keys_to_check_for, :changes, :errors
+      
+      def initialize
+        @changes = []
+        @errors = []
+        @metadata_map = {
+          "Bredig, Georg, 1868-" => "Bredig, Georg, 1868-1944",
+          "Caruso, David J."     => "Caruso, David J., (David Joseph), 1978-",
+        }
+        @keys_to_check_for = @metadata_map.keys
+      end
+
+      def process_work(work)
+        update_subject(work) unless (work.subject              & @keys_to_check_for).empty?
+        update_creator(work) unless (work.creator.map(&:value) & @keys_to_check_for).empty?
```

Comment:
Aha, I had missed this, sorry!

Reply:
This also let's us see if that if subject AND creator were modified... it'll do two separate saves. If this were something we were going to use a LOT in a high-volume way, we might want to avoid that... just for this simple utility on our actual data set? It's fine!

Reply:
Totally